samples/mlhelib_tutorial/tut_1_basics.cpp

```diff
@@ -0,0 +1,181 @@
+  // Later we'll show how this is useful.
+
+  // Let's create our first ciphertext.
+  // We'll first create a vector of doubles (1,2,3).
+  vector<double> vals1{1.0, 2.0, 3.0};
+
+  // To encrypt it, we need an encoder . . .
+  Encoder encoder(he);
+
+  // And a CTile object - this is our ciphertext object.
+  CTile c1(he);
+
+  // We'll now encrypt vals1 into c:
+  // In HE encryption actually involves two steps: encode, then encrypt.
+  // The following method does both.
+  encoder.encodeEncrypt(c1, vals1);
+
+  // c1 now contains the vector (1,2,3) encrypted. Since our library was
+  // configured
+  // to work with 8192 slots, it actually contains (1,2,3,0,0,0...0), with 0-s
+  // filling all unused slots.
+
+  // now let's create a second ciphertext:
+  vector<double> vals2{4.0, 5.0, 6.0};
+  CTile c2(he);
+  encoder.encodeEncrypt(c2, vals2);
+
+  // now let's add them together:
+  c1.add(c2);
+
+  // c1 is now the encryption of (5,7,9,0,0,0,...), resulting from adding
+  // (4,5,6,...) to (1,2,3,...).
+  // Actually, since the underlying scheme is CKKS, c1 is only approximately
+  // this result.
+  // I.e, instead of 5 we may get 5.0000001.
+
+  // Here are three ways to verify it.
+  // First, lets simply decrypt and see what we get.
+  // This also actually involves two steps: decrypt and decode, done by the
+  // following function.
+  // Also, some underlying schemes don't work directly with
+  // doubles.
+  // The one we chose, actually encrypts complex numbers.
+  // Since here we only use doubles, we chose the "decryptDecodeDouble" which
+  // automatically
+  // converts results back to doubles for us.
+  vector<double> res = encoder.decryptDecodeDouble(c1);
+  // Now let's print the first 4 elements. It should be close to 5,7,9,0.
+  cout << "After add:" << endl;
+  for (int i = 0; i < 4; ++i)
+    cout << "Result[" << i << "]=" << res[i] << endl;
+
+  // And here's a second way to verify it, useful for debugging, or for
+  // client-side tests.
+  // We rely here on the fact our HeContext object also has the secret
+  // (decryption) key.
+  // This is useful in the development stage, but in production it will only be
+  // available on client
+  // side (the client that generated the keys). On the server side, in
+  // production, this method
+  // won't work.
+  vector<double> expectedRes{5, 7, 9};
+  encoder.assertEquals(c1, "test result after add", expectedRes, 1e-5);
+  // The above call verified c1 contains 5,7, and  9 in the first 3 slots
+  // up-to precision of 1e-5. If not, an exception will be thrown.
+  cout << "\nAdd worked correctly!" << endl;
+
+  // And here's a third method relying access to the secret key.
+  // this will print the first 4 slots (as complex numbers) + the last slot +
+  // possibly some additional meta data of this ciphertext (depending on scheme)
+  c1.debugPrint("after add");
+
+  // Let's now quickly review more operators you can apply to ciphertexts.
```

Comment:
Can you gear the comments toward someone who might not know why we need to use this?  I think a non-crypto dev probably isn't sure why this is useful